internal/controller/compute/statefulserverset/datavolume_controller.go

```diff
@@ -0,0 +1,165 @@
+package statefulserverset
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	ionoscloud "github.com/ionos-cloud/sdk-go/v6"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/apis/compute/v1alpha1"
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/internal/controller/serverset"
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/internal/controller/volumeselector"
+)
+
+type kubeDataVolumeControlManager interface {
+	Create(ctx context.Context, cr *v1alpha1.StatefulServerSet, replicaIndex, volumeIndex int) (v1alpha1.Volume, error)
+	Get(ctx context.Context, volumeName, ns string) (*v1alpha1.Volume, error)
+	Delete(ctx context.Context, name, namespace string) error
+	Ensure(ctx context.Context, cr *v1alpha1.StatefulServerSet, replicaIndex, version int) error
+}
+
+// kubeDataVolumeController - kubernetes client wrapper  for server resources
+type kubeDataVolumeController struct {
+	kube client.Client
+	log  logging.Logger
+}
+
+// Create creates a volume CR and waits until in reaches AVAILABLE state
+func (k *kubeDataVolumeController) Create(ctx context.Context, cr *v1alpha1.StatefulServerSet, replicaIndex, volumeIndex int) (v1alpha1.Volume, error) {
```

Comment:
Why do we return `v1alpha1.Volume` in the response? I see that in the caller of this function, we only check the error

Reply:
Create functions usually return the object they create. Also helps if we need to test the function. Also that is what the other controllers and create functions in the repo do.

Reply:
Woudn't it be better to return a pointer instead of the actual structure? e.g. (*v1alpha1.Volume, error) ?